pkg/kubelet/kubelet.go

```diff
@@ -315,6 +315,9 @@ func NewMainKubelet(
 	}
 	klet.containerManager = containerManager
 
+	// Start syncing node status immediately, this may set up things the runtime needs to run.
+	go klet.syncNodeStatus()
```

Comment:
I don't think syncNodeStatus() immediately here will achieve what I what completely here. When node first joins the cluster, by default NodeStatus at nodecontroller is NotReady anyway, and master is waiting for kubelet sending Ready state, so that it can schedule pods to it. In that case, we shouldn't have problem with pods from apiserver.

When I suggested not to start any pods is really pointing to those file-based one. But SyncLoop never check if network is ready.